dbt_sugar/core/task/audit.py

```diff
@@ -0,0 +1,243 @@
+"""Document Task module."""
+from typing import Any, Dict, List
+
+from rich.console import Console
+from rich.table import Table
+
+from dbt_sugar.core.clients.yaml_helpers import open_yaml
+from dbt_sugar.core.flags import FlagParser
+from dbt_sugar.core.logger import GLOBAL_LOGGER as logger
+from dbt_sugar.core.task.base import COLUMN_NOT_DOCUMENTED, BaseTask
+
+console = Console()
+NUMBER_COLUMNS_TO_PRINT_PER_ITERACTION = 5
+
+
+class AuditTask(BaseTask):
+    """Audit Task object.
+
+    Holds methods and attrs necessary to audit a model or a DBT project.
+    """
+
+    def __init__(self, flags: FlagParser) -> None:
+        super().__init__()
+        self.column_update_payload: Dict[str, Dict[str, Any]] = {}
+        self._flags = flags
+        self.model_name = self._flags.model
+
+    def run(self) -> int:
+        """Main script to run the command doc"""
+        if self.model_name:
+            path_file, schema_exists = self.find_model_in_dbt(self.model_name)
+            if not path_file:
+                logger.info("Could not find the Model in the DBT project")
+                return 1
+            if not schema_exists:
+                logger.info("The model is not documented.")
+                return 1
+
+            self.model_content = open_yaml(path_file)
+            self.get_columns_descriptions_statistics_for_model()
+            self.get_columns_tests_statistics_for_model()
+        else:
+            self.get_all_columns_descriptions_statistics_per_model()
+            self.get_all_columns_tests_statistics_per_model()
+        return 0
```

Comment:
I feel the names here are so close to each other that it almost looks like dupes.

I think what this means is that in one case you kick in documentation for the model (i.e. you get the detailed coverage for all columns in a specific model and in the other case you're sumarising across a whole dbt project, am I correct?

Maybe what we could do on top of maybe clearer and more distinctive names would be to wrap those calls into dedicated runner functions such as:

```python
def derive_model_coverage():
    get_model_column_description_coverage()
    get_model_test_coverage()

def derive_project_coverage():
    get_project_column_description_coverage()
    get_project_test_coverage()
```

What do you think?

Reply:
really nice!